src/test/scala/io/iohk/ethereum/blockchain/sync/SyncControllerSpec.scala

```diff
@@ -523,59 +590,54 @@ class SyncControllerSpec extends AnyFlatSpec with Matchers with BeforeAndAfter w
-              if (underlyingMessage.maxHeaders == 1) {
-                // pivot block
-                sender ! MessageFromPeer(BlockHeaders(Seq(pivotHeader)), peer)
-              } else {
-                if (!onlyPivot) {
-                  val start = msg.underlyingMsg.block.left.get
-                  val stop = start + msg.underlyingMsg.maxHeaders
-                  val headers = (start until stop).flatMap(i => blockchainData.headers.get(i))
-                  sender ! MessageFromPeer(BlockHeaders(headers), peer)
-                }
-              }
-
-            case SendMessage(msg: GetReceiptsEnc, peer) if !onlyPivot =>
-              val underlyingMessage = msg.underlyingMsg
-              if (failedReceipts < failedReceiptsTries) {
-                sender ! MessageFromPeer(Receipts(Seq()), peer)
-                failedReceipts = failedReceipts + 1
-              } else {
-                val rec = msg.underlyingMsg.blockHashes.flatMap(h => blockchainData.receipts.get(h))
-                sender ! MessageFromPeer(Receipts(rec), peer)
-              }
-
-            case SendMessage(msg: GetBlockBodiesEnc, peer) if !onlyPivot =>
-              val underlyingMessage = msg.underlyingMsg
-              if (failedBodies < failedBodiesTries) {
-                sender ! MessageFromPeer(BlockBodies(Seq()), peer)
-                failedBodies = failedBodies + 1
-              } else {
-                val bod = msg.underlyingMsg.hashes.flatMap(h => blockchainData.bodies.get(h))
-                sender ! MessageFromPeer(BlockBodies(bod), peer)
-              }
+      testProbe.setAutoPilot(
+        SyncStateAutoPilot(
+          handshakedPeers,
+          pivotHeader,
+          blockchainData,
+          failedReceiptsTries,
+          failedBodiesTries,
+          onlyPivot,
+          failedNodeRequest
+        )
+      )
+    }
 
-            case SendMessage(msg: GetNodeDataEnc, peer) if !onlyPivot =>
-              stateDownloadStarted = true
-              val underlyingMessage = msg.underlyingMsg
-              if (!failedNodeRequest) {
-                sender ! MessageFromPeer(NodeData(Seq(defaultStateMptLeafWithAccount)), peer)
-              }
-          }
+    case class AutoPilotUpdateData(
+        handshakedPeers: HandshakedPeers,
+        pivotHeader: BlockHeader,
+        blockchainData: BlockchainData,
+        failedReceiptsTries: Int = 0,
+        failedBodiesTries: Int = 0,
+        onlyPivot: Boolean = false,
+        failedNodeRequest: Boolean = false
+    )
+    case object DataUpdated
 
-          this
-        }
-      })
+    def updateAutoPilot(
+        probeWIthPilot: ActorRef,
+        handshakedPeers: HandshakedPeers,
+        pivotHeader: BlockHeader,
+        blockchainData: BlockchainData,
+        failedReceiptsTries: Int = 0,
+        failedBodiesTries: Int = 0,
+        onlyPivot: Boolean = false,
+        failedNodeRequest: Boolean = false
+    ): Unit = {
+      val sender = TestProbe()
```

Comment:
Why not making it a method on autopilot? I can see convenience argument, but also in this way it would be easy to send a message to probe, which doesn't have autpilot installed